lib/iopromise/deferred/promise.rb

```diff
@@ -5,10 +5,14 @@
 module IOPromise
   module Deferred
     class DeferredPromise < ::IOPromise::Base
-      def initialize(&block)
+      def initialize(timeout: nil, &block)
```

Comment:
**Question** Perhaps I misunderstand the intention here, but reading the PR description, I would've expected this argument to be called `delay` or similar. I associate `timeout` with functionality that tries an operation _within the given timeout time_. The meaning here is different though?

Reply:
This is a great point :+1: